### PR TITLE
SSR: Experimental feature warnings

### PR DESCRIPTION
**Pull request by yopox**
Following discussions in #6675, this PR makes it clear that SSR is a work in progress.

changelog: Added an experimental feature to enable Rust SSR, added errors in the SSR dialog for unsupported templates.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt [hunk 1]
```diff
@@ -34,12 +54,69 @@ class RsStructuralSearchProfile : StructuralSearchProfile() {
...
+        physical: Boolean
+    ): Array<PsiElement> {
+        var patternTree = super.createPatternTree(text, context, fileType, language, contextId, project, physical)
+        if (patternTree.firstOrNull() is PsiErrorElement
+            || patternTree.size > 1 && patternTree[0] is LeafPsiElement && patternTree[1] is PsiErrorElement
+            || patternTree.firstOrNull() is DummyBlock) {
+            val factory = RsPsiFactory(project)
+            patternTree = factory.createBlockExpr(text).firstChild.children
+        }
+//        for (element in patternTree) print(DebugUtil.psiToString(element, false))
```
Comment (dima74) [inline src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt:82]:
```suggestion
```

File: src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt [hunk 2]
```diff
@@ -34,12 +54,69 @@ class RsStructuralSearchProfile : StructuralSearchProfile() {
...
+        project: Project,
+        physical: Boolean
+    ): Array<PsiElement> {
+        var patternTree = super.createPatternTree(text, context, fileType, language, contextId, project, physical)
+        if (patternTree.firstOrNull() is PsiErrorElement
+            || patternTree.size > 1 && patternTree[0] is LeafPsiElement && patternTree[1] is PsiErrorElement
+            || patternTree.firstOrNull() is DummyBlock) {
+            val factory = RsPsiFactory(project)
+            patternTree = factory.createBlockExpr(text).firstChild.children
+        }
```
Comment (dima74) [inline src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt:81]:
```suggestion
        val firstPattern = patternTree.firstOrNull()
        if (firstPattern is PsiErrorElement || firstPattern is DummyBlock
            || patternTree.size > 1 && patternTree[0] is LeafPsiElement && patternTree[1] is PsiErrorElement) {
            val factory = RsPsiFactory(project)
            patternTree = factory.createBlockExpr(text).firstChild.children
        }
```

File: src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt (hunk 1 above)
Comment (yopox) [reply src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt:82]: Can we leave it? This line is really really useful as a debug tool, it prints the template in the console for tests or in IDE

Comment (dima74) [reply src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt:82]: Ok let's keep

**Pull request merged**